### PR TITLE
threads: More descriptive startup output

### PR DESCRIPTION
**Pull request by JoshuaLumb**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Describe changes:
- re: https://github.com/OISF/suricata/pull/5192#pullrequestreview-450492183
- More descriptive output for thread creation.
- e.x. (Threads created -> RX: 1 W: 12 FM: 1 FR: 1)
- Now only enumerates thread counts > 0

### CODE REVIEW

File: src/tm-threads.c [hunk 1]
```diff
@@ -1873,18 +1877,67 @@ TmEcode TmThreadWaitOnThreadInit(void)
...
 
-    SCLogNotice("all %"PRIu16" packet processing threads, %"PRIu16" management "
-              "threads initialized, engine started.", ppt_num, mgt_num);
+    /* Construct a welcome string displaying
+     * initialized thread types and counts */
+    if (active_runmode < RUNMODE_USER_MAX) {
+        uint16_t app_len = 32;
+        uint16_t buf_len = 256;
+
+        char* append_str = SCMalloc(app_len);
```
Comment (victorjulien) [inline src/tm-threads.c:1904]: We can use the stack here to keep things simple `char append_str[app_len];` (same for threads_count)

File: src/tm-threads.c [hunk 2]
```diff
@@ -1873,18 +1877,67 @@ TmEcode TmThreadWaitOnThreadInit(void)
...
             tv = tv->next;
         }
     }
     SCMutexUnlock(&tv_root_lock);
 
-    SCLogNotice("all %"PRIu16" packet processing threads, %"PRIu16" management "
-              "threads initialized, engine started.", ppt_num, mgt_num);
+    /* Construct a welcome string displaying
+     * initialized thread types and counts */
+    if (active_runmode < RUNMODE_USER_MAX) {
```
Comment (victorjulien) [inline src/tm-threads.c:1900]: What is the purpose of this check?

File: src/tm-threads.c [hunk 3]
```diff
@@ -1873,18 +1877,67 @@ TmEcode TmThreadWaitOnThreadInit(void)
                 return TM_ECODE_FAILED;
             }
 
-            if (i == TVT_MGMT)
-                mgt_num++;
-            else if (i == TVT_PPT)
-                ppt_num++;
+            if (active_runmode < RUNMODE_USER_MAX) {
```
Comment (victorjulien) [inline src/tm-threads.c:1880]: what is the purpose of this check?

File: src/tm-threads.c (hunk 1 above)
Comment (JoshuaLumb) [reply src/tm-threads.c:1904]: Ah yeah good call! Thanks.

**Pull request closed without merging**